src/atdf/interrupt.rs

```diff
@@ -5,7 +5,17 @@ use crate::ElementExt;
 pub fn parse(interrupt: &xmltree::Element) -> crate::Result<chip::Interrupt> {
     interrupt.check_name("interrupt")?;
 
-    let name = interrupt.attr("name")?.clone();
+    let name = {
+        let inst_name = interrupt.attr("name")?;
+        interrupt
+            .attr("module-instance")
+            .map_or(inst_name.clone(), |s| {
```

Comment:
This will always clone the `inst_name`, whether needed or not.  I think it would be cleaner to use `.map_or_else()` and then only clone it when it is actually used:

```suggestion
            .map_or_else(|_| inst_name.clone(), |s| {
```

Reply:
Sure, didn't new about the eagerly evaluation of `map_or` :smile: